bench-exchange/README.md

```diff
@@ -59,76 +51,99 @@ matching trade orders.  All the transactions can execute concurrently.
      ratios are represented as fixed point numbers.  The fixed point scaler is
      defined in
      [exchange_state.rs](https://github.com/solana-labs/solana/blob/c2fdd1362a029dcf89c8907c562d2079d977df11/programs/exchange_api/src/exchange_state.rs#L7)
-- Trade request
-  - A Solana transaction executed by the exchange requesting the trade of one
-    type of token for another.  Trade requests are made up of the token pair,
-    the direction of the trade, quantity of the primary token, the price ratio,
-    and the two token accounts to be credited/deducted.  An example trade
-    request looks like "T AB 5 2" which reads "Exchange 5 A tokens to B tokens
-    at a price ratio of 1:2"  A fulfilled trade would result in 5 A tokens
-    deducted and 10 B tokens credited to the trade initiator's token accounts.
-    Successful trade requests result in a trade order.
-- Trade order
-  - The result of a successful trade request.  Trade orders are stored in
-    accounts owned by the submitter of the trade request.  They can only be
-    canceled by their owner but can be used by anyone in a trade swap.  They
-    contain the same information as the trade request.
+- Order Requests
+  - A Solana transaction sent by the owner of an exchange account and executed
+    by the exchange, resulting in the creation of an Order. It has these components:
+    - Offer - the contract address of an asset and an amount of the asset for sale by the maker
+    - Accept - the contract address of a asset and an amount of the asset accepted as compensation by the maker
+    - Expiration - a unix timestamp denoting the time of expiration of the order
+    - OrderOptions - optional parameters for trade execution/settlement behavior
+- Offer/Accept
+  - Are used to specify the terms of an Order. The order issuer elects to offer
+    certain assetAmounts and to accept as others as payment
+  - An AssetAmount (improve name) consists of an asset as identified by its contract address and an amount
+    of that asset in base units
+  - The Offer and Accept fields should be vectors populated by between 1 and n AssetAmounts
+    where n is a small number (10?)
+  - for efficient handling and execution, should be 1:1, 1:n, or n:1, not n:n
```

Comment:
Anything but 1:1 seems like it has good odds to conceptually snowball on us.  I'm not aware of this 1:N offer notion in any existing trading system and it's not clear what problem it tries to solve.

Reply:
The point of these multi-asset orders is to improve the marginal utility of capital for traders in a way that is intrinsic to the exchange and does not require external rebates or grant programs to incentivize market makers. They aren't an MVP feature.
As long as n is small the increase in resources used per order should be minimal since the validation and execution logic should all be O(n), and it will increase the efficiency of the market overall.

Reply:
basically the idea is that you capture the advantage of being able to to place more orders than you have capital like in a non-custodian dex but without the normal downside of that which is worrying about orders being funded. 
so you might place an order offering 1 sol and accepting [10 usdc, 10 gusd, 15 libra]
here, you have 3x the value of your offer in order(s?) outstanding, but because: 
 - the order is on chain and it's parameters update in the event of a partial fill
 - the total transact amount is escrowed
The parties taking/matching that order don't have to worry about it being unfillable any more than a typical 1:1 offer:accept order.